app/addons/databases/actions.js

```diff
@@ -0,0 +1,105 @@
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+define([
+  'app',
+  'api',
+  'addons/databases/stores',
+  'addons/databases/actiontypes',
+  'addons/databases/resources'
+],
+function (app, FauxtonAPI, Stores, ActionTypes, Resources) {
+  return {
+    init: function (collection, backboneCollection) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.DATABASES_INIT,
+        options: {
+          collection: collection,
+          backboneCollection: backboneCollection
+        }
+      });
+    },
+
+    setPage: function (_page) {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.DATABASES_SETPAGE,
+        options: {
+          page: _page
+        }
+      });
+    },
+
+    setStartLoading: function () {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.DATABASES_STARTLOADING,
+        options: {}
+      });
+    },
+
+    setLoadComplete: function () {
+      FauxtonAPI.dispatch({
+        type: ActionTypes.DATABASES_LOADCOMPLETE,
+        options: {}
+      });
+    },
+
+    createNewDatabase: function (databaseName, nameAccCallback) {
+      databaseName = (databaseName === null ? null : databaseName.trim());
+      if (databaseName === null || databaseName.length === 0) {
+        FauxtonAPI.addNotification({
+          msg: 'Please enter a valid database name',
+          type: 'error',
+          clear: true
+        });
+        return;
+      }
+      // name accepted, make sure prompt can be removed
+      nameAccCallback();
+
+      var db = Stores.databasesStore.obtainNewDatabaseModel(databaseName);
+      FauxtonAPI.addNotification({ msg: 'Creating database.' });
+      db.save().done(function () {
+          FauxtonAPI.addNotification({
+            msg: 'Database created successfully',
+            type: 'success',
+            clear: true
+          });
+          var route = '#/database/' + app.utils.safeURLName(databaseName) + '/_all_docs?limit=' + Resources.DocLimit;
+          app.router.navigate(route, { trigger: true });
+        }
+      ).error(function (xhr) {
+          var responseText = JSON.parse(xhr.responseText).reason;
+          FauxtonAPI.addNotification({
+            msg: 'Create database failed: ' + responseText,
+            type: 'error',
+            clear: true
+          });
+        }
+      );
+    },
+
+    jumpToDatabase: function (databaseName) {
+      databaseName = databaseName === null ? null : databaseName.trim();
+      if (databaseName === null || databaseName.length === 0) {
+        return;
+      }
+      if (Stores.databasesStore.getDatabaseNames().indexOf(databaseName) >= 0) {
```

Comment:
Could you make this a function on the databasesStore. Something like `Stores.databasesStore.doesDatabaseExist(databaseName)`